### PR TITLE
Replace tag references with SHAs for github actions

### PR DESCRIPTION
**Pull request by tylerbenson**
https://github.blog/changelog/2021-01-21-github-actions-short-sha-deprecation/

In the process, I also updated the various actions to their latest versions.

### CODE REVIEW

**Comment by devinsba** [comment]
Unfortunately the this misses the most important actions. The ones the in `actions/` namespace are owned by github so there is a level of trust there. We pull in a few third party actions that are far more important

File: .github/workflows/increment-milestones-on-tag.yaml [hunk 1]
```diff
@@ -51,13 +51,13 @@ jobs:
...
         if: fromJSON(steps.milestone.outputs.result)
         id: semvers
-        uses: WyriHaximus/github-action-next-semvers@0.1.0
+        uses: WyriHaximus/github-action-next-semvers@33d116a4c239252582a60a1ba8dbba63ad493ffd # 1.1.0
         with:
           version: ${{steps.milestoneTitle.outputs.result}}
       - name: Create next milestone
         if: fromJSON(steps.milestone.outputs.result)
-        uses: WyriHaximus/github-action-create-milestone@0.1.0
+        uses: WyriHaximus/github-action-create-milestone@b86699ba7511fa3b61154ac8675d86b01938fc16 # 1.0.0
```
Comment (tylerbenson) [inline .github/workflows/increment-milestones-on-tag.yaml:59]: @devinsba these are the only two I saw...

Comment (tylerbenson) [reply .github/workflows/increment-milestones-on-tag.yaml:59]: oops... now I see more. Thanks for catching that.

**Review by bantonsson — APPROVED** [review]

**Review by devinsba — APPROVED** [review]
I do have some concerns about the fact that some of these are major version changes from what we were using

**Comment by tylerbenson** [comment]
Yeah, I also have concern about upgrading, but I figured now is a good time to do so while we're looking closely at it.  We can roll back if it doesn't work.

**Comment by bantonsson** [comment]
I say let's upgrade since at least `WyriHaximus/github-action-next-semvers@0.1.0` is broken right now https://github.com/DataDog/dd-trace-java/runs/1928377046?check_suite_focus=true

**Pull request merged**